lightstep/resource_metric_dashboard.go

```diff
@@ -150,14 +150,27 @@ func getTextPanelSchema() map[string]*schema.Schema {
 }
 
 // getPanelSchema returns the common metadata on any dashboard panel (timeseries charts or text panels)
-func getPanelSchema() map[string]*schema.Schema {
-	return map[string]*schema.Schema{
-		// Alias for what we refer to as title elsewhere
-		"name": {
+//
+// Timeseries charts requires "name" to be a required, while text panels don't.
+func getPanelSchema(isNameRequired bool) map[string]*schema.Schema {
+	nameSchema := func() *schema.Schema {
+		if isNameRequired {
+			return &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			}
+		}
+
+		return &schema.Schema{
 			Type:     schema.TypeString,
 			Optional: true,
 			Default:  "",
-		},
+		}
+	}
```

Comment:
Nit: is a closure needed here?  Would it be more straightforward to put the logic inline?

```go
	return map[string]*schema.Schema{
		// Alias for what we refer to as title elsewhere
		"name": &schema.Schema{
			Type:     schema.TypeString,
			Required: isNameRequired,
            Optional: !isNamedRequired,
			Default:  "",
        },
        ...
```

Reply:
Yeah, I tried that, but if `Optional: true` we can't set anything to `Default`.